### ISSUE TITLE
'The "rest" option does not exist message' in console while running horizon

### ISSUE THREAD

**Issue by jishnulbit** [carsguide/lumen-horizon#12]
Add this fix to the code. https://github.com/laravel/horizon/blob/5.x/CHANGELOG.md#fixed-2

**Comment by medbak** [comment]
I have the same issue and I tried to fix it but I didn't the origin of this issue, but I can say it is not working with Lumen 8.

**Comment by Rasoul-Karimi** [comment]
There are no commands defined in the "vendor" namespace.  
is not working with lumen

**Comment by lenartjakub** [comment]
It' s doesn't work beacuse in WorkCommand from package not there added flag --rest. I made this like that:

- create new HorizonServiceProvider extended by HorizonServiceProvider from carsguide package,
- Copied whole method regeisterCommands(), but I deleted WorkCommand::class from parent class. 
- I made own WorkCommand in project and added missing flag. My attribute signature look like this:
`protected $signature = 'horizon:work
                            {connection? : The name of the queue connection to work}
                            {--name=default : The name of the worker}
                            {--delay=0 : The number of seconds to delay failed jobs (Deprecated)}
                            {--backoff=0 : The number of seconds to wait before retrying a job that encountered an uncaught exception}
                            {--max-jobs=0 : The number of jobs to process before stopping}
                            {--max-time=0 : The maximum number of seconds the worker should run}
                            {--daemon : Run the worker in daemon mode (Deprecated)}
                            {--force : Force the worker to run even in maintenance mode}
                            {--memory=128 : The memory limit in megabytes}
                            {--once : Only process the next job on the queue}
                            {--stop-when-empty : Stop when the queue is empty}
                            {--queue= : The names of the queues to work}
                            {--sleep=3 : Number of seconds to sleep when no job is available}
                            {--supervisor= : The name of the supervisor the worker belongs to}
                            {--timeout=60 : The number of seconds a child process can run}
                            {--rest=0 : Number of seconds to rest between jobs}
                            {--tries=0 : Number of times to attempt a job before logging it failed}';`
- last step is add my own WorkCommand::class, where previous we deleted that row.

Dont' forget implements your HorizonProviderService in bootstrap/app.php

**Comment by medbak** [comment]
Thanks Jakub,

I tried as you say, so provider looks like 
![image](https://user-images.githubusercontent.com/10911162/134524705-b83f2943-d6c1-4b86-b50f-c4170b9345f9.png)

 workcommand
![image](https://user-images.githubusercontent.com/10911162/134524872-aaffab7c-c97e-4d5c-95d8-c9933ac049ef.png)

and bootstrap/app.php
![image](https://user-images.githubusercontent.com/10911162/134524976-071e6654-9d21-45a6-aef5-186c8e09de6e.png)

when I executed the command php artisan horizon, I got this error message
![image](https://user-images.githubusercontent.com/10911162/134525133-be834e58-a6e1-40cb-a8d0-1125c967320d.png)
`In Container.php line 1104:
                                                                                                                                   
  Unresolvable dependency resolving [Parameter #3 [ <required> callable $isDownForMaintenance ]] in class Illuminate\Queue\Worker `

Am I missing something ?

**Comment by medbak** [comment]
> It' s doesn't work beacuse in WorkCommand from package not there added flag --rest. I made this like that:
> 
>     * create new HorizonServiceProvider extended by HorizonServiceProvider from carsguide package,
> 
>     * Copied whole method regeisterCommands(), but I deleted WorkCommand::class from parent class.
> 
>     * I made own WorkCommand in project, copied $signature attribute from WorkCommand and added missing flag:
>       `{--rest=0 : Number of seconds to rest between jobs}`
> 
>     * last step is add my own WorkCommand::class, where previous we deleted that row.
> 
> 
> Dont' forget implements your HorizonProviderService in bootstrap/app.php

Thanks Jakub,

Would you please see if I am missing something ?  I add the steps below.
Thanks again.

**Comment by lenartjakub** [comment]
@medbak you are right. I forgot about one case. Add to your HorizonServiceProvider that method:
```
    public function register(): void
    {
        $this->app->bind(WorkCommand::class, function ($app) {
            return new WorkCommand($app['queue.worker'], $app['cache.store']);
        });

        $this->configure();
        $this->offerPublishing();
        $this->registerServices();
        $this->registerCommands();
        $this->registerQueueConnectors();
    }
```
Probably it was the reason of error.

**Comment by medbak** [comment]
> @medbak you are right. I forgot about one case. Add to your HorizonServiceProvider that method:
> 
> ```
>     public function register(): void
>     {
>         $this->app->bind(WorkCommand::class, function ($app) {
>             return new WorkCommand($app['queue.worker'], $app['cache.store']);
>         });
> 
>         $this->configure();
>         $this->offerPublishing();
>         $this->registerServices();
>         $this->registerCommands();
>         $this->registerQueueConnectors();
>     }
> ```
> 
> Probably it was the reason of error.

Thanks @lenartjakub,

I added the function in HorizonServiceProvider but  still get an issue 

`In QueueManager.php line 172:
                        
  No connector for [].  
`
![image](https://user-images.githubusercontent.com/10911162/134641167-da429657-0333-4c35-b657-71fca58e596a.png)

I am sure that the configuration of the queue and cache with redis it is working fine, I verified Horizon config and It looks fine to me 

![image](https://user-images.githubusercontent.com/10911162/134641356-3ce43ce8-d95e-4b95-8991-972196d5a60a.png)

**Comment by Arielpod** [comment]
I created a merge request that fixes this issue as explained by @jishnulbit.
